rocAL/rocAL/source/parameters/parameter_rali_crop.cpp

```diff
@@ -58,8 +58,19 @@ void RocalCropParam::fill_crop_dims()
             // Evaluating user given crop
             (crop_w > in_width[img_idx]) ? (cropw_arr_val[img_idx] = in_width[img_idx]) : (cropw_arr_val[img_idx] = crop_w);
             (crop_h > in_height[img_idx]) ? (croph_arr_val[img_idx] = in_height[img_idx]) : (croph_arr_val[img_idx] = crop_h);
-            (x1 >= in_width[img_idx]) ? (x1_arr_val[img_idx] = 0) : (x1_arr_val[img_idx] = x1);
-            (y1 >= in_height[img_idx]) ? (y1_arr_val[img_idx] = 0) : (y1_arr_val[img_idx] = y1);
+            if(_is_center_crop)
+            {
+                float x_drift, y_drift;
+                x_drift = x_drift_factor->get();
+                y_drift = y_drift_factor->get();
+                x1_arr_val[img_idx] = static_cast<size_t>(x_drift * (in_width[img_idx]  - cropw_arr_val[img_idx]));
+                y1_arr_val[img_idx] = static_cast<size_t>(y_drift * (in_height[img_idx] - croph_arr_val[img_idx]));
+            }
+            else
+            {
+                (x1 >= in_width[img_idx]) ? (x1_arr_val[img_idx] = 0) : (x1_arr_val[img_idx] = x1);
+                (y1 >= in_height[img_idx]) ? (y1_arr_val[img_idx] = 0) : (y1_arr_val[img_idx] = y1);
```

Comment:
The above 2 lines can also be changed this way

x1_arr_val[img_idx] = (x1 >= in_width[img_idx]) ? 0 : x1;

Also line number 95 and 96 can be changed in a similar way